Public/Src/Tools/SymbolDaemon/VsoSymbolClient.cs

```diff
@@ -124,11 +132,42 @@ public async Task<AddDebugEntryResult> AddFileAsync(SymbolFile symbolFile)
 
             await EnsureRequestIdInitalizedAsync();
 
-            var result = await m_symbolClient.CreateRequestDebugEntriesAsync(
-                RequestId,
-                symbolFile.DebugEntries.Select(e => CreateDebugEntry(e)),
-                DefaultDebugEntryCreateBehavior,
-                CancellationToken);
+            List<DebugEntry> result;
+
+            try
+            {
+                result = await m_symbolClient.CreateRequestDebugEntriesAsync(
+                    RequestId,
+                    symbolFile.DebugEntries.Select(e => CreateDebugEntry(e)),
+                    DebugEntryCreateBehavior.ThrowIfExists,
+                    CancellationToken);
+            }
+            catch (DebugEntryExistsException)
+            {
+                string message = $"[SymbolDaemon] File: '{symbolFile.FullFilePath}' caused collision. " +
+                    (m_debugEntryCreateBehavior == DebugEntryCreateBehavior.ThrowIfExists 
+                        ? string.Empty 
+                        : $"SymbolDaemon will retry creating debug entry with {m_debugEntryCreateBehavior} behavior");
+
+                // Log a warning message in BuildXL log file
+                Analysis.IgnoreResult(await m_apiClient.LogMessage(message, isWarning: true));
+ 
+                if (m_debugEntryCreateBehavior == DebugEntryCreateBehavior.ThrowIfExists)
+                {
+                    // Log an error message in SymbolDaemon log file
+                    m_logger.Error(message);
+                    throw new DebugEntryExistsException(message);

```

Comment:
>throw new DebugEntryExistsException(message); [](start = 20, length = 45)

just `throw;`

Reply:
I add a new message to the exception; as the existing message is too confusing. That's why, I did not want to use 'throw'.